stubs/requests/requests/sessions.pyi

```diff
@@ -48,11 +48,23 @@ class SessionRedirectMixin:
 _Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | None]] | IO[Any]
 _Auth: TypeAlias = Union[tuple[str, str], _auth.AuthBase, Callable[[PreparedRequest], PreparedRequest]]
 _Cert: TypeAlias = Union[str, tuple[str, str]]
+# Files can be passed as either a mapping, or a list of tuples.
+_FileContent: TypeAlias = SupportsRead[str | bytes] | str | bytes
+_FileName: TypeAlias = str | None
+_FileContentType: TypeAlias = str
+_FileCustomHeaders: TypeAlias = _TextMapping
+_FileSpecTuple2 = tuple[_FileName, _FileContent]
+_FileSpecTuple3 = tuple[_FileName, _FileContent, _FileContentType]
+_FileSpecTuple4 = tuple[_FileName, _FileContent, _FileContentType, _FileCustomHeaders]
 _Files: TypeAlias = (
-    Mapping[str, SupportsRead[str | bytes] | str | bytes]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes]]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str]]
-    | Mapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, _TextMapping]]
+    Mapping[str, _FileContent]
+    | Mapping[str, _FileSpecTuple2]
+    | Mapping[str, _FileSpecTuple3]
+    | Mapping[str, _FileSpecTuple4]
+    | Sequence[tuple[str, _FileContent]]
+    | Sequence[tuple[str, _FileSpecTuple2]]
+    | Sequence[tuple[str, _FileSpecTuple3]]
+    | Sequence[tuple[str, _FileSpecTuple4]]
```

Comment:
I don't think we need to arduously spell these out as separate kinds of mappings. From the source code, it looks like each `key, value` pair in `files` is checked individually to see what the structure is: https://github.com/psf/requests/blob/a5e7169e759817ca309f8271c3c4fa79393a22f5/requests/models.py#L177-L186.

Local testing appears to confirm that "mixing and matching" different structures in the same request seems to work fine, e.g.

```python
import requests

url = 'https://httpbin.org/post'
multiple_files = [
    ('images', ('foo.png', open('foo.png', 'rb'))),
    ('images', ('bar.png', open('bar.png', 'rb'), 'image/png')),
    ('images', ('baz.png', open('baz.png', 'rb'), 'image/png', {}))
]
r = requests.post(url, files=multiple_files)
print(r.status_code)
print(r.text)
```

As such, we can take advantage of covariance, and do this instead:

```python
_OneOrSeveralFiles: TypeAlias = _FileContent | _FileSpecTuple2 | _FileSpecTuple3 | _FileSpecTuple4

_Files : TypeAlias = (
    Mapping[str, _OneOrSeveralFiles]
    | Sequence[tuple[str, _OneOrSeveralFiles]]
)
```